manifests/server/root_password.pp

```diff
@@ -19,6 +19,12 @@
   $secret_file = $mysql::server::install_secret_file
   $login_file = $mysql::server::login_file
 
+  # Validate that the given secret_file value is a valid path
+  if $secret_file !~ /^(?:\/[A-Za-z0-9._-]+)+?$/ {
```

Comment:
I think this regex needs some work, we might actually want to parse this with a ruby function if it's exposed instead of trying to regex it. Maybe something like:

```
secret_file = Shellwords.escape($secret_file)
if(File.file?($secret_file))
... 
```

Reply:
Agree that the regex is a bit shady..

shell_escape is exposed via stdlib so we can use that to clean the input.

I did want some sort of validation  happen though.. unless that what File.file? would provide in your example?

Reply:
We cannot do actual filesystem validation because this runs during compile only. The best we can do is string validation that it's a valid pathname.

One thing we could do for this class of problem is to adopt my [assert resource](https://github.com/binford2k/puppet-assert) into stdlib and conditionally skip resources if assertions (like File.file?) were to fail. That's probably a different conversation though.

Reply:
This code has been removed now, so I'm resolving this.